### DISCUSSION TITLE
Arabic is Displaying in Reverse

### DISCUSSION THREAD

**Discussion by astreffect** [dompdf/dompdf#3266] [Q&A] [answered]
Hi I was trying to create a form to pdf using dompdf but Am facing the issue that the Arabics generatd alongside english is left to right and not rtl, if I force rtl by editing the dompdf text.php then the english words also reverse also the fact that arabics are joined words and in dompdf words are seperate.

Can you please help me out with :
1. proper rtl display of arabic words
2. Joined/combined arabic words instead of seperate
3. Make sure English & Arabics can co-exist.

I am attaching relevant part of my code :

```
use Dompdf\Dompdf;
use Dompdf\Options;

if (isset($_POST['generate-pdf'])) {
    $options = new Options();
    $options->set('isHtml5ParserEnabled', true);
    $options->set('isPhpEnabled', true);
    $options->set('fontDir', __DIR__ . '/fonts'); // Path to your fonts directory
    $options->set('fontCache', __DIR__ . '/font-cache'); // Path to your font cache directory
    $options->set('defaultFont', 'DejaVuSans');

    $dompdf = new Dompdf($options);

    $template = file_get_contents('pdf-template.html');

    $template = str_replace('<head>', '<head><style>.arabic { direction: rtl; font-family: DejaVuSans; }</style>', $template);

    // Map form fields to placeholders in the template
    $template = str_replace('{{date}}', $_POST['date'], $template);
    $template = str_replace('{{company_name}}', $_POST['company_name'], $template);
     // Convert the image to base64
    $logo_base64 = base64_encode(file_get_contents('alent.PNG'));
    $template = str_replace('{{logo_base64}}', $logo_base64, $template);

    $html = $template;


    $dompdf->loadHtml($html);
    $dompdf->setPaper('A4', 'portrait');
    $dompdf->render();
    $dompdf->stream('output.pdf', array('Attachment' => 0));
}
```

```
<!DOCTYPE html>
<html><html lang="ar" dir="rtl">
<head>
    <meta charset="UTF-8">

    <title>PDF Template</title>
    <style>

        /* Add your CSS styles here */
        .header {
            display: flex;
            align-items: center;
        }
        @font-face {
            font-family: 'DejaVu Sans';
            src: url('/fonts/DejaVuSans.ttf') format('truetype'),
            url('/fonts/DejaVuSans.woff') format('woff');
            font-weight: normal;
            font-style: normal;
        }

        body {
            font-family: 'DejaVu Sans', sans-serif, Arial;
        }
        body {
            direction: rtl; /* Right-to-Left */
        }

        h3
        {
            size: A4;
            text-align: right;
            font-size: 16px;
        }

        .logo {
            max-width: 100px;
            margin-right: 20px;
        }
```
        
        
Hope to get a solution soon.

**Comment by TimexPeachtree** [comment, marked as answer]
Please look at this issue hope this solves the issue did the work for me more than 10 years using the solutions provided here

[DOMPDF - problem in Arabic text Issue 712 - Solved Gold](https://github.com/dompdf/dompdf/issues/712)


Also Ar-PHP author Mr. Khaled Al Shamaa gave the solution with installing new version of AR-PHP on same issue
[AR-PHP New version](https://github.com/dompdf/dompdf/issues/712#issuecomment-650592099)

**Reply by astreffect** [reply]
Thanks, it did work like magic.

**Comment by bsweeney** [comment]
Dompdf does not currently support rtl or some other language features required to render Arabic text. We had started work on supporting this functionality in #2107, but have not yet been able to resolve some issues related to multi-line text and lines containing multiple text frames (e.g. inline elements). Until we are able to work on the functionality further the resolution provided in #712 (as indicated by TimexPeachtree) is your best bet.
